client-mr/src/main/java/org/apache/hadoop/mapreduce/task/reduce/RssShuffle.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Tencent is pleased to support the open source community by making
+ * Firestorm-Spark remote shuffle server available.
+ *
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.hadoop.mapreduce.task.reduce;
+
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.tencent.rss.client.api.ShuffleReadClient;
+import com.tencent.rss.client.api.ShuffleWriteClient;
+import com.tencent.rss.client.factory.ShuffleClientFactory;
+import com.tencent.rss.client.request.CreateShuffleReadClientRequest;
+import com.tencent.rss.common.ShuffleServerInfo;
+import com.tencent.rss.common.util.UnitConverter;
+import org.apache.hadoop.mapred.*;
+import org.apache.hadoop.mapreduce.MRRssUtils;
+import org.apache.hadoop.mapreduce.RssMRConfig;
+import org.apache.hadoop.mapreduce.TaskAttemptID;
+import org.apache.hadoop.util.Progress;
+import org.apache.hadoop.yarn.api.ApplicationConstants;
+import org.apache.hadoop.yarn.api.records.ContainerId;
+import org.roaringbitmap.longlong.Roaring64NavigableMap;
+
+public class RssShuffle<K, V> implements ShuffleConsumerPlugin<K, V>, ExceptionReporter {
```

Comment:
RssShuffleConsumer？

Reply:
In original MR， this class is just called `Shuffle` (does not ask me why).

Reply:
OK, although a little strange.